Description: Fix ReDoS vulnerability in archetype lexer
 The AdlLexer's GUID and ID patterns could be exploited with crafted input
 to cause excessive CPU usage via inefficient regular expression complexity.
 .
 This patch limits the length of repeated patterns to prevent ReDoS attacks.
Author: zam. <zamkorong12@gmail.com>
Origin: upstream, https://github.com/pygments/pygments/commit/24b8aa76c6cd6d70f39c6dd605cce319c98e2ccc
Bug: https://security-tracker.debian.org/tracker/CVE-2026-4539
Bug-Debian: https://bugs.debian.org/1132233
Forwarded: not-needed

Index: github-pygments-CVE-2026-4539/pygments/lexers/archetype.py
===================================================================
--- github-pygments-CVE-2026-4539.orig/pygments/lexers/archetype.py
+++ github-pygments-CVE-2026-4539/pygments/lexers/archetype.py
@@ -39,8 +39,8 @@ class AtomsLexer(RegexLexer):
             (r'([ \t]*)(--.*)$', bygroups(Whitespace, Comment)),
         ],
         'archetype_id': [
-            (r'([ \t]*)(([a-zA-Z]\w+(\.[a-zA-Z]\w+)*::)?[a-zA-Z]\w+(-[a-zA-Z]\w+){2}'
-             r'\.\w+[\w-]*\.v\d+(\.\d+){,2}((-[a-z]+)(\.\d+)?)?)',
+            (r'([ \t]*)(([a-zA-Z]\w{1,100}(\.[a-zA-Z]\w{1,100})*::)?[a-zA-Z]\w{1,100}(-[a-zA-Z]\w{1,100}){2}'
+             r'\.\w{1,100}[\w-]*\.v\d+(\.\d+){,2}((-[a-z]+)(\.\d+)?)?)',
              bygroups(Whitespace, Name.Decorator)),
         ],
         'date_constraints': [
@@ -297,7 +297,7 @@ class AdlLexer(AtomsLexer):
             # numbers and version ids
             (r'\d+(\.\d+)*', Literal),
             # Guids
-            (r'(\d|[a-fA-F])+(-(\d|[a-fA-F])+){3,}', Literal),
+            (r'[0-9a-fA-F]{1,36}(-[0-9a-fA-F]{1,36}){3,}', Literal),
             (r'\w+', Name.Class),
             (r'"', String, 'string'),
             (r'=', Operator),
